src/internal/SignalGenerator.cpp

```diff
@@ -0,0 +1,25 @@
+#include <bits/stdc++.h>
+#include <hyro/SignalGenerator.h>
+
+namespace hyro 
+{
+
+SignalGenerator::SignalGenerator()
+  : m_deg(0.0)
+  , m_step(0.0)
+  , m_amp(0.0)
+{
+}
+
+double SignalGenerator::getSignalValue()
+{
+  m_rad = m_deg / 180.0 * M_PI;
```

Comment:
`m_rad` is just used here. It can be defined locally, instead of being a member of the class.

```
double rad = m_deg / 180.0 * M_PI;
```